skopt/optimizer/optimizer.py

```diff
@@ -166,6 +168,11 @@ def _check_arguments(self, base_estimator, n_random_starts, acq_optimizer):
             warnings.warn("The 'auto' option for the acq_optimizer will be "
                           "removed in 0.4.")
             acq_optimizer = "lbfgs"
+
+        if isinstance(base_estimator, (DecisionTreeRegressor,
+                                       ExtraTreesRegressor)):
+            acq_optimizer = "sampling"
```

Comment:
I'd prefer it if this raised an exception telling the user that they picked an invalid combination of arguments instead of surprising them by resetting an argument.